### ISSUE TITLE
Defining custom datadir ineffective on Windows

### ISSUE THREAD

**Issue by leafcutterant** [ethereum/go-ethereum#2625]
#### System information

Geth version: `1.4.5 (Drop table users)`
OS & Version: Windows 7 x64
Commit hash : - (if `develop`)

#### Expected behaviour

Geth should work with the custom defined datadir when started
1. with the `--datadir "<datadirpath>"` from command line, or
2. (and this is just an estimation from Bitcoin command syntax) with a shortcut and defining the Target as `"<gethpath>" -datadir="<datadirpath>"`

#### Actual behaviour

1. Geth starts up acknowledging the new datadir, but then quickly produces a database error and nothing happens afterwards.
`C:\Users\****>E:\Programs\Ethereum\Geth\geth.exe --datadir "G:\Data\Ethereum"
I0526 11:43:43.504181 ethdb/database.go:82] Alloted 128MB cache and 1024 file handles to G:\Data\Ethereum\chaindata
Fatal: Could not open database: leveldb/table: corruption on table-footer (pos=2115192): bad magic number [file=099345.ldb]
Fatal: Could not open database: leveldb/table: corruption on table-footer (pos=2115192): bad magic number [file=099345.ldb]`

2. Geth starts up acknowledging the new datadir (`I0526 11:51:44.600698 ethdb/database.go:82] Alloted 128MB cache and 1024 file handles to G:\Data\Ethereum\chaindata`), but no synchronization happens, then after one minute of standing by, it exits unexpectedly.

#### Steps to reproduce the behaviour

Copy the datadir to a new location and start Geth with the above methods, using the path of the new location.

#### Backtrace

````
[backtrace]
````

Sorry, I may be wrong on how to define the Windows shortcut target, but there was not much info available.

**Comment by bas-vk** [comment]
Could you verify that the data copied is not modified? This error indicates that the new datadir is corrupt and something went wrong during copying. You can use `--datadir` and `-datadir`, both work.

One thing that is odd in your first bullet. You specify a datadir on the G drive and according to the logging geth seems to open the database on the F drive. Did you edit something?

**Comment by leafcutterant** [comment]
Okay, I apologize, there was a mess-up. The copy was indeed corrupted, made a new one, re-started, and it's working. I don't really understand why. My guess is that during the initial copy Geth was still running in the background and files were not in the same state and/or some couldn't be copied over.

Did a little experimenting, I can confirm that synchronization successfully starts with both syntactical forms (`<gethpath> --datadir <datadirpath>` and `"<gethpath>" -datadir="<datadirpath>"`). These were in the shortcut target, they probably work in command line too.

Yes, I edited the log because meanwhile I swapped my drives, but apparently I missed a letter...

Sorry once again for the false alarm, closing.
